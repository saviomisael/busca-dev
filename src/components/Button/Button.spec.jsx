import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Button } from '.';

describe('<Button />', () => {
  const onClickMock = jest.fn();

  it('should renders component with default props', () => {
    global.renderTheme(<Button onClick={onClickMock}>Button</Button>);

    const button = screen.getByRole('button', { name: 'Button' });

    expect(button).toHaveStyleRule('width', '100%');
  });

  it('should renders component with medium size', () => {
    global.renderTheme(
      <Button onClick={onClickMock} size="medium">
        Button
      </Button>,
    );

    const button = screen.getByRole('button', { name: 'Button' });

    expect(button).toHaveStyleRule('width', '85%');
  });

  it('should be more brightness when button hover', () => {
    global.renderTheme(
      <Button onClick={onClickMock} size="medium">
        Button
      </Button>,
    );

    const button = screen.getByRole('button', { name: 'Button' });

    expect(button).toHaveStyleRule('filter', 'brightness(1.2)', {
      modifier: ':hover',
    });
  });

  it('should goes down when button is clicked', () => {
    global.renderTheme(
      <Button onClick={onClickMock} size="medium">
        Button
      </Button>,
    );

    const button = screen.getByRole('button', { name: 'Button' });

    expect(button).toHaveStyleRule('transform', 'translateY(0.8rem)', {
      modifier: ':active',
    });
  });

  it('should call onClick function when button is clicked', () => {
    global.renderTheme(
      <Button onClick={onClickMock} size="medium">
        Button
      </Button>,
    );

    const button = screen.getByRole('button', { name: 'Button' });

    userEvent.click(button);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
