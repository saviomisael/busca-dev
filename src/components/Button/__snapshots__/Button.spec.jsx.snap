// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should matches with snapshot 1`] = `
.c0 {
  text-transform: uppercase;
  padding: 0.8rem 1.5rem;
  border-radius: 0.8rem;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  font-size: 1.8rem;
  background: #353535;
  color: #F5F5F5;
  width: 85%;
}

.c0:hover {
  -webkit-filter: brightness(1.2);
  filter: brightness(1.2);
}

.c0:active {
  -webkit-transform: translateY(0.8rem);
  -ms-transform: translateY(0.8rem);
  transform: translateY(0.8rem);
}

<button
  class="c0"
>
  Button
</button>
`;
